final/master.py

```diff
@@ -0,0 +1,408 @@
+                    current_red = window.get_at((x, y)).r
+                    current_green = window.get_at((x, y)).g
+                    current_blue = window.get_at((x, y)).b
+
+                    red_total += current_red
+                    green_total += current_green
+                    blue_total += current_blue
+                    colour_count += 1
+
+            # Have to swap red and blue here sometimes... No idea why. Maybe png and jpg render bgr not rgb
+            new_red = red_total / colour_count
+            new_green = green_total / colour_count
+            new_blue = blue_total / colour_count
+
+            pixel_array[currentStartX: currentStartX + tile_size - tile_spacing,
+                currentStartY: currentStartY + tile_size - tile_spacing] = (new_red, new_green, new_blue)
+
+    pict = pixel_array.make_surface()
+    del pixel_array
+    del px_array
+
+    window.blit(pict, (0, 0))
+    pygame.display.update()
+
+
+def print_controls():
+    print ('Controls:')
+    print ('Q - Invert colours')
+    print ('W - Blur')
+    print ('E - Night vision')
+    print ('R - Waterfall')
+    print ('T - Horizontal lines, length is similarity strength')
+    print ('Y - Circles')
+    print ('U - Outline')
+    print ('I - Horizontal lines, fading, chosen by strength')
+    print ('O - Simplify colours')
+    print ('P - Pixelise, with tile effect')
+    print ('L - P then O combo. Works well.')
+    print ('Space - Reset')
+
+print_controls()
+
+
+while True:
+    keys = pygame.key.get_pressed()
+    px_array = pygame.PixelArray(window)
+    clock = pygame.time.Clock()
+    clock.tick(10)
+    for event in pygame.event.get():
+        if event.type == QUIT:
+            pygame.quit()
+            sys.exit()
+
+    # Blurs When 'w' is pressed
+    if keys[pygame.K_w]:
+        blur_picture(a)
+
+    # Inverts colours when 'q' is pressed
+    if keys[pygame.K_q]:
+        color_invert()
+
+    # Night Vision when e is pressed
+    if keys[pygame.K_e]:
+        night_vision()
+
+    # Press r for vertical lines
+    if keys[pygame.K_r]:
+        water_fall()
+
+    # Press t for horizontal lines
+    if keys[pygame.K_t]:
+        horizontal_lines()
+
+    # Press y for circles
+    if keys[pygame.K_y]:
+        circle()
+
+    # Adds outline
+    if keys[pygame.K_u]:
+        outline()
+
+    # Fading streak effect comparison
+    if keys[pygame.K_i]:
+        streaker()
+
+    # Simplifies the colours
+    if keys[pygame.K_o]:
+        simplify_colour()
+
+    # Pixelise picture in interesting ways
+    if keys[pygame.K_p]:
+        pixelise_is_a_word_stop_underlining_it_in_grey()
+
+    # Good combos
+    if keys[pygame.K_l]:
+        pixelise_is_a_word_stop_underlining_it_in_grey()
+        px_array = pygame.PixelArray(window)
+        simplify_colour()
+
+    # Blits original picture
+    if keys[pygame.K_SPACE]:
+        del px_array
+        window.blit(picture, (0, 0))
+        run_once = True
```

Comment:
What is the purpose of `run_once`? This is not clear. Provide a comment, or refactor, as appropriate.